searchlib/src/vespa/searchlib/tensor/hnsw_index.cpp

```diff
@@ -415,16 +436,26 @@ HnswIndex::filter_valid_nodeids(uint32_t level, const PreparedAddDoc::Links &nei
 void
 HnswIndex::internal_complete_add(uint32_t docid, PreparedAddDoc &op)
 {
-    auto nodeids = _id_mapping.allocate_ids(docid, 1);
-    assert(nodeids.size() == 1);
-    auto nodeid = nodeids[0];
-    auto node_ref = _graph.make_node(nodeid, docid, 0, op.max_level + 1);
-    for (int level = 0; level <= op.max_level; ++level) {
-        auto neighbors = filter_valid_nodeids(level, op.connections[level], nodeid);
+    assert(op.nodes.size() == 1);
+    auto nodeids = _id_mapping.allocate_ids(docid, op.nodes.size());
+    assert(nodeids.size() == op.nodes.size());
+    uint32_t subspace = 0;
+    for (auto nodeid : nodeids) {
+        internal_complete_add_node(nodeid, docid, subspace, op.nodes[subspace]);
```

Comment:
I guess subspace should be increased as well?

Reply:
Yes.